ui/src/features/stage/promotion-steps-wizard/promotion-steps-wizard.tsx

```diff
@@ -0,0 +1,174 @@
+                  className={classNames('cursor-pointer', {
+                    'shadow-sm': isEditingThisRunner,
+                    'border-gray-400': isEditingThisRunner
+                  })}
+                  size='small'
+                  onClick={onSettingOpen}
+                >
+                  <div className='flex items-center gap-5'>
+                    <div className='flex flex-col gap-5 text-2xl cursor-pointer'>
+                      {order > 0 && (
+                        <FontAwesomeIcon
+                          icon={faCaretUp}
+                          onClick={(e) => {
+                            e.preventDefault();
+                            e.stopPropagation();
+                            const newOrder = [...selectedRunners];
+                            newOrder[order] = selectedRunners[order - 1];
+                            newOrder[order - 1] = runner;
+                            setSelectedRunners(newOrder);
+                            setEditingRunner({ ...runner, order: order - 1 });
+                          }}
+                        />
+                      )}
+                      {order < selectedRunners.length - 1 && (
+                        <FontAwesomeIcon
+                          icon={faCaretDown}
+                          onClick={(e) => {
+                            e.preventDefault();
+                            e.stopPropagation();
+                            const newOrder = [...selectedRunners];
+                            newOrder[order] = selectedRunners[order + 1];
+                            newOrder[order + 1] = runner;
+                            setSelectedRunners(newOrder);
+                            setEditingRunner({ ...runner, order: order + 1 });
+                          }}
+                        />
+                      )}
+                    </div>
+
+                    <span className='font-semibold'>
+                      {order + 1} - {runner.identifier}
+                    </span>
+
+                    {!!runner?.as && (
+                      <Tag className='text-xs max-w-20 overflow-hidden ml-auto' color='blue'>
+                        {runner.as}
+                      </Tag>
+                    )}
+
+                    <div className={classNames('space-x-4', { 'ml-auto': !runner?.as })}>
+                      <Button icon={<FontAwesomeIcon icon={faCog} />} onClick={onSettingOpen} />
+                      <Button
+                        icon={<FontAwesomeIcon icon={faTrash} className='text-red-500' />}
+                        onClick={(e) => {
+                          e.stopPropagation();
+                          setSelectedRunners(
+                            selectedRunners.filter((_, deleteOrder) => deleteOrder !== order)
+                          );
+                        }}
+                      />
+                    </div>
+                  </div>
+                </Card>
+              );
```

Comment:
Can we extract this into a separate component?

Reply:
Done